xarray/conventions/netcdftimeindex.py

```diff
@@ -120,23 +118,31 @@ def get_date_type(self):
     return type(self._data[0])
 
 
-def assert_all_same_netcdftime_datetimes(data):
-    from netcdftime._netcdftime import datetime
+def assert_all_valid_date_type(data):
+    from netcdftime import (
```

Comment:
You can just use datetime here -- these are all subclasses, so isinstance checks on the super class work fine.